### PR TITLE
Release: 4.0.0-RC.4

### PR DESCRIPTION
**Pull request by aboedo**
- Fourth RC for RevenueCat framework v4 🎉
    100% Swift framework + ObjC support.

[Full Changelog](https://github.com/revenuecat/purchases-ios/compare/4.0.0-rc.3...4.0.0-rc.4)
- See our [RevenueCat V4 API update doc](Documentation.docc/V4_API_Migration_guide.md) for API updates.

RC 4 introduces the following updates:

#### Breaking changes: 
- Replaced `checkPromotionalDiscountEligibility` with `getPromotionalOffer`, which returns a `PromotionalOffer`. 
- Renamed `Purchases/purchase(package:discount:)` and its variants to `Purchases/purchase(package:promotionalOffer:)`. They now take a `PromotionalOffer` instead of a `StoreProductDiscount`.
- [Objective-C only]: Updated type of `StoreProduct.price` and `StoreProductDiscount.price` from `NSDecimal` to the much more useful `NSDecimalNumber`. 

#### Additions:
- Added `StoreProduct.ProductType`, and `StoreProduct.ProductCategory`, which provide extra information about whether a product is a consumable, non-consumable, auto-renewable or non-auto-renewable subscription.
- Added `currencyCode` to `StoreProduct` and `StoreProductDiscount`.
- Added `localizedPriceString` to `StoreProductDiscount`.

### Documentation: 

- Documentation can be found in https://revenuecat-docs.netlify.app/documentation/Revenuecat. 
- We've made several improvements to docstrings and added a few landing pages for the most important sections of the SDK. 

### Other changes: 
- There are lots of under the hood improvements. If you see any issues we'd appreciate [bug reports](https://github.com/RevenueCat/purchases-ios/issues/new?assignees=&labels=bug&template=bug_report.md&title=)!

### Changes from previous RC

These changes add to all of the changes from beta RC 2, [listed here.](https://github.com/RevenueCat/purchases-ios/blob/4.0.0-rc.3/CHANGELOG.latest.md).

### CODE REVIEW

**Comment by aboedo** [comment]
This is another instance where the changelog.latest file is the only part really worth reviewing, everything else is auto-generated

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+
+These changes add to all of the changes from beta RC 2, [listed here.](https://github.com/RevenueCat/purchases-ios/blob/4.0.0-rc.3/CHANGELOG.latest.md).
+
 ## 4.0.0-RC.3
 
 - Third RC for RevenueCat framework v4 🎉
     100% Swift framework + ObjC support.
 
-[Full Changelog](https://github.com/revenuecat/purchases-ios/compare/4.0.0-beta.rc.2...4.0.0-rc.3)
+[Full Changelog](https://github.com/revenuecat/purchases-ios/compare/4.0.0-rc.2...4.0.0-rc.3)
```
Comment (aboedo) [inline CHANGELOG.md:40]: the link was wrong here

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,7 +69,7 @@ These changes add to all of the changes from beta RC 2, [listed here.](https://g
 - Second RC for RevenueCat framework v4 🎉
     100% Swift framework + ObjC support.
 
-[Full Changelog](https://github.com/revenuecat/purchases-ios/compare/4.0.0-beta.rc.1...4.0.0-rc.2)
+[Full Changelog](https://github.com/revenuecat/purchases-ios/compare/4.0.0-rc.1...4.0.0-rc.2)
```
Comment (aboedo) [inline CHANGELOG.md:72]: same here, I updated the actual GH releases

File: CHANGELOG.latest.md [hunk 3]
```diff
@@ -1,29 +1,31 @@
-- Third RC for RevenueCat framework v4 🎉
```
Comment (aboedo) [inline CHANGELOG.latest.md:1]: This is the only file really worth reviewing

**Review by taquitos — APPROVED** [review]
🕺 🐐 🥳

File: IntegrationTests/CommonFiles/RevenueCat-Swift.h [hunk 4]
```diff
@@ -1929,7 +2077,8 @@ enum RCDiscountType : NSInteger;
 SWIFT_CLASS_NAMED("StoreProductDiscount")
 @interface RCStoreProductDiscount : NSObject
 @property (nonatomic, readonly, copy) NSString * _Nullable offerIdentifier;
-@property (nonatomic, readonly) NSDecimal price;
```
Comment (NachoSoto) [inline IntegrationTests/CommonFiles/RevenueCat-Swift.h:1932]: Nice, no longer here :P

**Review by NachoSoto — APPROVED** [review]

**Pull request merged**